test/distance/edge-filters-test.cpp

```diff
@@ -0,0 +1,115 @@
+#include <gtest/gtest.h>
+
+#include "test/common/common.hpp"
+#include "src/distance/edge.hpp"
+#include "src/distance/edge-filters.hpp"
+#include "src/providers/factory.hpp"
+
+namespace found {
+
+TEST(EdgeFiltersTest, EdgeFilteringAlgorithms_RunWithoutCompletionThrows) {
```

Comment:
This is already guarenteed per the `pipeline` tests. No need for duplicate tests